pkg/cli/admin/upgrade/status/workerpool.go

```diff
@@ -0,0 +1,506 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	mcfgv1client "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	mcfgcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	mcfgconst "github.com/openshift/machine-config-operator/pkg/daemon/constants"
```

Comment:
The size of the vendored changes was bugging me so I followed that rabbit hole and the result is https://github.com/openshift/oc/pull/1697, check it out.

I discovered that we are vendoring o/m-c-o for basically two purposes, nothing else:
1. Constants
2. `LayeredNodeState` we use for `IsUnavailable`

We can inline the constants (or better, copy the files that contain them into a little `pkg/cli/admin/upgrade/status/mco/constants.go` ). The `LayeredNodeState` code is a little harder, but still fairly self-contained, check out https://github.com/openshift/oc/pull/1697/commits/1a61719ca0adba097b571f20d1c73e408a20045e. We can hardcopy two files (say, again, into `pkg/cli/admin/upgrade/status/mco/...`) and delete the bits we do not use, resulting in very small part of the o/m-c-o code tht we actually use, and *then* all we need for deps is `go mod vendor` which simply brings the missing parts of client-go and o/api - without modifying go.sum/go.mod at all.

This would be a little heavy-handed, but remember all this client-side code is experimental and will be eventually removed, and as long as we comment the extracted MCO code with prominent `TODO: replace this with proper MCO code when going server-side`, I think it is worth it (avoids bringing a ton of new deps for code that will never reach customers).

WDYT?

Reply:
Seems very reasonable. I will proceed with this suggestion. Thank you very much, I appreciate the extensive explanation.